app/components/SettingsCellComponent.js

```diff
@@ -0,0 +1,120 @@
+import React from 'react'
+import {
+  View,
+  Text,
+  StyleSheet,
+  TouchableOpacity,
+  Dimensions,
+  Switch,
+} from 'react-native'
+import { colors } from '../config'
+
+const { settingsCellTextColor, settingsCellBorderLineColor } = colors
+const { width: screenWidth } = Dimensions.get('screen')
+
+const SettingsCellComponent = ({
+  cellText,
+  onCellPress,
+  onSwitchValueChange,
+  isSwitchCellComponent,
+}) => {
+  const cellPressed = () => {
+    if (onCellPress) {
+      onCellPress()
+    }
+  }
+
+  const switchValueChanged = () => {
+    if (onSwitchValueChange) {
+      onSwitchValueChange()
+    }
+  }
+
+  const textComponent = () => {
+    const { textContainerStyle, textStyle } = styles
+    return (
+      <View style={textContainerStyle}>
+        <Text style={textStyle}>{cellText}</Text>
+      </View>
+    )
+  }
+
+  const switchOrArrowComponent = () => {
+    const { switchOrArrowContainerStyle } = styles
+    return (
+      <View style={switchOrArrowContainerStyle}>
+        {isSwitchCellComponent ? (
+          <Switch onValueChange={switchValueChanged} />
+        ) : null}
+      </View>
+    )
+  }
+
+  const textOnArrowCell = () => {
+    const { touchableOpacityStyle } = styles
+    return (
+      <TouchableOpacity style={touchableOpacityStyle} onPress={cellPressed}>
+        {textComponent()}
+      </TouchableOpacity>
+    )
+  }
+
+  const getCellComponent = () => {
+    const { subContainerStyle } = styles
+    return (
+      <View style={subContainerStyle}>
+        {isSwitchCellComponent ? textComponent() : textOnArrowCell()}
+        {switchOrArrowComponent()}
+      </View>
```

Comment:
I don't like this approach. These could by used in the return (pretty basic JSX these functions are returning) if it were more complex break it out into a component.

Reply:
yeah i had changed the approach as u suggested.